bindings/packager/nodejs/src-ts/plugins/index.ts

```diff
@@ -0,0 +1,56 @@
+import path from "path";
```

Comment:
if we are going to add plugins, should we add one for tauri as well? and probably write all in Rust?

Reply:
if I wrote all this in Rust you would tell me not to bloat the cargo packager with NPM stuff 😂 
idk about Tauri, i think most users will stick to tauri-bundler since they are compatible. the problem is just the renaming of the app right? (plus it's harder to detect a tauri project instead of an electron app since Node.js helps us there)

Reply:
It is not really about bloat right now, just trying to understand the big picture here, if we are adding plugins, what plugins in the future we might have? why electron gets a special plugin, why dioxus or wry can't have a plugin for it?

Reply:
Electron is completely different because it's precompiled so it goes totally against the concepts of the packager, you need to ship a bunch of resources and files by default. It's not 100% controlled by the developer.

Reply:
> Electron is completely different because it's precompiled so it goes totally against the concepts of the packager

That doesn't really go against it, the packager does expect a prebuilt binary

> you need to ship a bunch of resources and files by default.

From packager POV, that's just some configurations that can be set manually by the developer, we are just making it easier for them which I am not against, just seeing if we would do the same for tauri and dioxus.

Reply:
Tauri or Dioxus AFAIK does not need any external files to work, but Electron needs you to include their libraries too. If we didn't have this "plugin", every Electron dev would need to download the prebuilt app on their own, figure out the format and what files to include. It's a nightmare and something we can do on our end.
If there's another framework that need this, I think it's ok to add it in the future too. As long as it's easy to detect.

Reply:
> Tauri or Dioxus AFAIK does not need any external files to work,

They do need specific configurations for gtk libraries and `deb` depdendencies and webview2 installation

> If there's another framework that need this, I think it's ok to add it in the future too. As long as it's easy to detect.

yeah that's what I am trying to figure out so we can decide how to make it future-proof, will the packager include plugins or a preconfigured options for specific frameworks to make better DX?